packages/contracts/src/2.0.0/protocol/Exchange/libs/LibAbiEncoder.sol

```diff
@@ -0,0 +1,215 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental ABIEncoderV2;
+
+import "./LibOrder.sol";
+
+
+contract LibAbiEncoder {
+   
+    /// @dev ABI encodes calldata for `fillOrder` in memory and returns the address range.
+    ///      This range can be passed into `call` or `delegatecall` to invoke an external
+    ///      call to `fillOrder`.
+    /// @param order Order struct containing order specifications.
+    /// @param takerAssetFillAmount Desired amount of takerAsset to sell.
+    /// @param signature Proof that order has been created by maker.
+    /// @return calldataBegin Memory address of ABI encoded calldata.
+    /// @return calldataLength Lenfgth of ABI encoded calldata.
+    function abiEncodeFillOrder(
+        LibOrder.Order memory order,
+        uint256 takerAssetFillAmount,
+        bytes memory signature,
+        bytes4 fillOrderSelector
```

Comment:
Let's just hard code the selector into the function, like we do in the other areas we ABI encode data in assembly.